### PR TITLE
Add ability to create `.copywrite.hcl` file in downstreams

### PR DESCRIPTION
**Pull request by SarahFrench**
**Context:**

HashiCorp has a tool called [copywrite](https://github.com/hashicorp/copywrite) that adds licenses and copyright headers to files in public HashiCorp repos. There's also automation where PRs will be opened against repos that have any of that info missing, e.g.:
- https://github.com/hashicorp/terraform-provider-google/pull/13624
- https://github.com/hashicorp/terraform-provider-google-beta/pull/5155

**This PR:**

This PR adds a config file that influences the behaviour of this automation/actions of the [copywrite CLI](https://github.com/hashicorp/copywrite).

By defining a .copywrite.hcl file in Magic Modules we can ensure the downstream google and google-beta repos have the same config in place.

The intention of this PR is to settling on which files we want excluded from copyright headers being added to.
In a future PR(s) we can add automation to make sure adding the headers is part of the process for generating the downstream.

---

If this PR is for Terraform, I acknowledge that I have:

- [x] ~~Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).~~
- [x] ~~Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).~~
- [x] ~~[Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.~~
- [x] ~~[Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).~~
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/.copywrite.hcl.erb:18]: This is a config folder for a HashiCorp internal tool - this folder doesn't exist in the downstream repos yet but I thought I'd include it here in preparation for when it may be added in future

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/.copywrite.hcl.erb:16]: This folder isn't affected by the copywrite CLI tool as all the files are `.txt`. However if we changed to a different way of generating changelogs, e.g. using yaml files, then copywrite could cause a lot of trouble there.

The comment on L12 is about this item.

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/.copywrite.hcl.erb:17]: Generally, files in the examples folder can be used to generate provider documentation using the `tfplugindocs` package. We don't use this for the Google provider, but I thought it was worth preventing files being edited in case we swap over in future.

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/.copywrite.hcl.erb:15]: This directory has been floated as something to ignore, as they're important to the release process. Need to confirm.

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/.copywrite.hcl.erb:19]: This file has been floated as something to ignore. Need to confirm.

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/.copywrite.hcl.erb:20]: This file has been floated as something to ignore. Need to confirm.

**Review by SarahFrench — COMMENTED** [review]
Self-review to help document my thinking for each entry and any uncertainty about entries.

File: mmv1/third_party/terraform/.copywrite.hcl.erb
Comment (dlaguerta) [reply mmv1/third_party/terraform/.copywrite.hcl.erb:19]: I think we'll leave it up to teams to use their best judgement when it comes to ignoring some of these configurations, like `.golangci.yml`. It's definitely safe to ignore as long as it doesn't include any IP.

**Comment by SarahFrench** [comment]
👆 I've got the ok from the HashiCorp side, so now Hashi/Google maintainers are free to discuss without any unknowns

**Review by roaks3 — APPROVED** [review]
Changes LGTM

My understanding based on the downstream diffs is that this will create the `.copywrite.hcl` files, and then copywrite will create PRs for adding the actual headers. Is that correct? If it is I have 2 follow-up questions:
- Do we need to worry about the headers being overwritten by magic-modules?
- Is there any way to see the end result from these changes (ie. which files are modified), or is the plan to wait for copywrite to run and adjust accordingly?

**Comment by SarahFrench** [comment]
Good questions - first, yes, this PR is just to add a file to each of the downstreams and when those automated PRs are opened/updated the file will influence which files have headers added to them.

>Do we need to worry about the headers being overwritten by magic-modules?

Yes we do. At the moment I'm not approving the automated PRs that I linked in the description and it's likely that they'll be closed eventually. I figured an easy first step is to add these config files to the downstreams and have any necessary discussions about which files should be excluded now. After that we can add the copywrite CLI tool to our automation for building the downstreams. [I have a PR in draft for this](https://github.com/GoogleCloudPlatform/magic-modules/pull/7255) but I'm not sure how to test the changes - I will be talking to someone on the Google team about this in the near future!

The eventual goal is we have this CLI added to our Magic Modules automation and then no more ["[COMPLIANCE]" PRs](https://github.com/hashicorp/terraform-provider-google/pull/13624) will be opened in future, because that HashiCorp automation won't find any changes to suggest.

>Is there any way to see the end result from these changes (ie. which files are modified), or is the plan to wait for copywrite to run and adjust accordingly?

To test the changes I [installed the copywrite CLI using Homebrew](https://github.com/hashicorp/copywrite#getting-started) and ran `copywrite headers` in my cloned version of the TPG repo. The CLI also has a `--plan` flag that is useful, and the config file in this PR is used automatically because it has the default name.

**Comment by SarahFrench** [comment]
Also, I'll avoid merging this PR until it's discussed in the next weekly sync. It might be that merging is delayed until the PR adding copywrite to the MM automation is closer to being finished.

**Comment by SarahFrench** [comment]
Just making a note for myself: need to add info about the upstream repo, using this property in the config: https://github.com/hashicorp/copywrite/blob/2fb1cc17e1fc1b5edd24e4d595af9abad944ba7e/README.md?plain=1#L113-L116

**Comment by SarahFrench** [comment]
Updated the file with that info about upstream repos, and rebased the PR

**Comment by SarahFrench** [comment]
I've prepared 3 PRs that allow copyright headers to be added to TPG/TPGB repos from Magic Modules and 2 are approved so I'm going to get started merging them. But first I'm merging this PR 🎉 

This is my plan: https://github.com/hashicorp/terraform-provider-google/pull/13624#issuecomment-1562821920

**Pull request merged**
